deploy.cmd

```diff
@@ -89,20 +89,26 @@ goto :EOF
 echo Handling node.js deployment.
 
 
-:: 1. Select node version
+echo 1. Select node version
 call :SelectNodeVersion
 
-:: 2. Install npm packages
+echo  2. Install npm packages
 IF EXIST "%DEPLOYMENT_SOURCE%\package.json" (
   pushd "%DEPLOYMENT_SOURCE%"
-  call :ExecuteCmd !NPM_CMD! install --production
+  call :ExecuteCmd !NPM_CMD! install
   IF !ERRORLEVEL! NEQ 0 goto error
   popd
 )
-:: 3. Angular Prod Build
-IF EXIST "%DEPLOYMENT_SOURCE%/.angular.json" (
+echo 3. Angular Prod Build
+IF EXIST "%DEPLOYMENT_SOURCE%/angular.json" (
 echo Building App in %DEPLOYMENT_SOURCE%…
 pushd "%DEPLOYMENT_SOURCE%"
+cd ./node_modules/@angular/cli
+rd /s /q node_modules
+call !NPM_CMD! install
+cd D:\home\site\repository
```

Comment:
I'm also slightly puzzled by this -- why do we need to do an `npm install` inside this folder as well?

Reply:
ikr??? Angular is missing a bunch of modules for the build. Here is one of the thread's I got the idea from [https://github.com/angular/angular-cli/issues/1782#issuecomment-245240123]. Without this piece, you get a seemingly-never-ending stream of missing module failures. Hoping we can integrate upgrades to Angular to eventually replace this.

Reply:
I don't understand. The comment you refer to talks about running `npm install portfinder` to fix the broken dependency.

Reply:
So that's just one of the missing modules for angular, but the general principle is still the same. Here's another [thread](https://github.com/angular/angular-cli/issues/1186)